pkg/api/pin_chunks.go

```diff
@@ -312,10 +318,34 @@ func (s *server) updatePinCount(ctx context.Context, reference swarm.Address, de
 
 func (s *server) pinChunkAddressFn(ctx context.Context, reference swarm.Address) func(address swarm.Address) (stop bool) {
 	return func(address swarm.Address) (stop bool) {
-		err := s.Storer.Set(ctx, storage.ModeSetPin, address)
+		// NOTE: stop pinning on first error
+
+		has, err := s.Storer.Has(ctx, address)
+		if err != nil {
+			s.Logger.Debugf("pin traversal: localstore has: for reference %s, address %s: %w", reference, address, err)
```

Comment:
Note about this comment.

At first I did make the suggested change, I have since moved it back to `Debug` level. Reason for this is because on any error the iteration will be stopped abruptly, and I think that `Debug` level is useful, as only single line will be written. If the case would be that for each missing chunk we log something I would move it to `Trace` level.

If you still feel we should move it to `Trace` level, please comment, and I will add the change.

Reply:
OK fair enough :+1:
My reasoning was because: when logging actions on individual chunks I tend to stick to Trace but if we're talking about 1 log line that will be written and then the operation terminates then all good :+1: